lib/mm_input.rb

```diff
@@ -1,16 +1,13 @@
-HELP = 'HELP'
-QUIT = 'QUIT'
-EXIT = 'EXIT'
-RULES = 'RULES'
+require_relative 'mm_outputs'
 
 class Input
-    attr_reader :user_input, :is_valid, :output_code, :restart_request, :show_help, :show_rules, :err_data
+    attr_reader :user_input, :is_valid, :output_code, :restart_request, :show_help, :show_rules, :err_data, :cmd_run
 
-    def initialize
+    def initialize(out = Outputs)
+        @out = Outputs
         @show_help = false
         @show_rules = false
-        puts 'Please enter a code or command...'
-        print "---> "
+        @out.in_prompt
```

Comment:
Could you pull out the prompting/validating into a function and call it after initialization/not part of it?